src/util/earthEngine.js

```diff
@@ -0,0 +1,164 @@
+import i18n from '@dhis2/d2-i18n';
+import { formatStartEndDate } from './time';
+import { loadEarthEngineApi } from '../components/map/MapApi';
+import { apiFetch } from './api';
+import { getEarthEngineLayer } from '../constants/earthEngine';
+
+export const classAggregation = ['percentage', 'hectares', 'acres'];
+
+export const hasClasses = type => classAggregation.includes(type);
+
+export const getStartEndDate = data =>
+    formatStartEndDate(
+        data['system:time_start'],
+        data['system:time_end'], // - 7200001, // Minus 2 hrs to end the day before
+        null,
+        false
+    );
+
+export const getPeriodFromFilter = (filter = []) => {
+    const f = filter[0];
+
+    return f
+        ? {
+              id: f.id || f.arguments[1],
+              name: f.name,
+              year: f.year,
+          }
+        : null;
+};
+const setAuthToken = async ({ client_id, access_token, expires_in }) =>
+    new Promise((resolve, reject) => {
+        ee.data.setAuthToken(client_id, 'Bearer', access_token, expires_in);
+        ee.initialize(null, null, resolve, reject);
+    });
+
+// Set token and load api
+const connectEarthEngine = () =>
+    new Promise(async (resolve, reject) => {
+        const token = await apiFetch('/tokens/google').catch(() =>
+            reject({
+                type: 'engine',
+                error: true,
+                message: i18n.t(
+                    'Cannot get authorization token for Google Earth Engine.'
+                ),
+            })
+        );
+
+        if (token && token.status === 'ERROR') {
+            reject({
+                type: 'engine',
+                warning: true,
+                message: i18n.t(
+                    'This layer requires a Google Earth Engine account. Check the DHIS2 documentation for more information.'
+                ),
+            });
+        }
+
+        if (!window.ee && loadEarthEngineApi) {
+            await loadEarthEngineApi();
+        }
+
+        try {
+            await setAuthToken(token);
+        } catch (e) {
+            reject({
+                type: 'engine',
+                error: true,
+                message: i18n.t('Cannot connect to Google Earth Engine.'),
+            });
+        }
+
+        resolve(window.ee);
+    });
+
+const getInfoPromise = obj => new Promise(resolve => obj.getInfo(resolve));
+
+export const getPeriods = async id => {
+    const { periodType } = getEarthEngineLayer(id);
+    const ee = await connectEarthEngine();
+
+    const imageCollection = ee
+        .ImageCollection(id)
+        .distinct('system:time_start')
+        .sort('system:time_start', false);
+
+    const featureCollection = ee
+        .FeatureCollection(imageCollection)
+        .select(['system:time_start', 'system:time_end'], null, false);
+
+    // Return first and last date for daily periods
+    if (periodType === 'Daily') {
```

Comment:
Should these be constants imported from somewhere?

Reply:
Kept for now, but I removed the code for Daily period as it's not in use yet.